[sniff] Bracket Placement

dcolt opened the pull request:
Difference from original:

1. Added `try` token to the list to register
2. Added fix for comment on the same line

dcolt committed:
[sniff] Bracket Placement

bix0r reviewed (commented):
I am not sure what we are trying to do here. But, I don't agree that we should allow comments before constrol structures.

Also, should this not check and fix:
```if (false) {
  doStuff();
} else {
  doOtherStuff();
}```
=> ```if (false) {
  doStuff();
} 
else {
  doOtherStuff();
}```

sunkan commented on tests/Sniffs/ControlStructures/data/BracketPlacementSniff.OK.php:
```diff
@@ -0,0 +1,21 @@
+for ($i = 0; $i < 1; $i++) {} 
+// End For
+if (false) {} 
+/* End If */ elseif (false) {} 
```
I'm with biggi on this these comments should definitely not be allowed

I on the other hand think these should be allowed
```
if (false) {
} 
elseif (true) { // comment here
}

```

dcolt committed:
[refactor] Use TokenCollection in BracketPlacementSniff

dcolt committed:
[tests,BracketPlacement] Format OK, add test case

dcolt committed:
[sniff, BracketPlamentSniff] Rewritten rules

Re-written the rules as following:
1. Control statements has to be the first non-whitespace on the line.
2. Closing bracket must be the first non-whitespace on the line (Except
   for empty scopes)

dcolt commented:
So I've re-written the rules to hopefully be a bit more clear on what we attempt to do.

The new rules is my understanding of what we tried to do with the previous variant.
See https://github.com/stefna/codestyle/pull/10/commits/e1a63c26f26c09f1ae6fbe1839d0dd1a20d04808 for new definitions of the rules.

dcolt committed:
[test] Added missing string from OK

dcolt committed:
[tests] Updated tests to be split instead of a single big test

dcolt committed:
[docs] Add BracketPlacementSniff into the readme
